### PR TITLE
857 sms model overhaul

### PR DESCRIPTION
**Pull request by linesarefuzzy**

### CODE REVIEW

File: app/mailers/broadcast_mailer.rb [hunk 1]
```diff
@@ -4,6 +4,6 @@ class BroadcastMailer < ActionMailer::Base
   def broadcast(recips, subj, msg)
     s = configatron.broadcast_tag + " " + (subj || "")
```
Comment (hooverlunch) [inline app/mailers/broadcast_mailer.rb]: This line is bit legacy/nasty. Could be moved into 7 and cleaned up.

File: app/models/broadcast.rb [hunk 2]
```diff
@@ -48,50 +46,56 @@ def email_possible?
...
 
-  def no_possible_recipients?
-    recipients.all?{ |u| u.email.blank? && u.phone.blank? && u.phone2.blank? }
+  def recipient_emails
+    @recipient_emails ||= recipients.map { |r| r.email if r.can_get_email? }.compact
   end
 
   private
 
     def has_eligible_recipients
```
Comment (hooverlunch) [inline app/models/broadcast.rb]: Should end in ?

File: db/migrate/20141211205915_migrate_sms_message_data.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    SQL
+
+    remove_column :sms_messages, :direction
+    change_column :sms_messages, :to, :string
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
+end
```
Comment (hooverlunch) [inline db/migrate/20141211205915_migrate_sms_message_data.rb]: Wow, nice work here. Had no idea how tricky that would be.

File: spec/models/sms/adapters/frontline_sms_adapter_spec.rb [hunk 4]
```diff
@@ -35,11 +35,11 @@
   it 'should correctly parse a frontline-style request' do
     Time.zone = ActiveSupport::TimeZone['Saskatchewan']
 
-    request = {'frontline' => '1', 'text' => 'foo', 'from' => '+2348036801489'}
+    request = {'frontline' => '1', 'text' => 'foo', 'from' => '+2348036801489', 'to' => '+123456789'}
```
Comment (hooverlunch) [inline spec/models/sms/adapters/frontline_sms_adapter_spec.rb]: I don't think the to will be included in the request. It should be taken from the mission SMS to number if given. If not, it should be allowed to be null. Both scenarios should probably be tested.

File: spec/models/sms/adapters/intelli_sms_adapter_spec.rb [hunk 5]
```diff
@@ -34,12 +34,12 @@
     Time.zone = ActiveSupport::TimeZone['Saskatchewan']
 
     request = {'text' => 'foo', 'sent' => '2013-07-03T09:53:00+01:00', 'from' => '2348036801489',
-      'msgid' => '1234'}
+      'msgid' => '1234', 'to' => '123456789'}
```
Comment (hooverlunch) [inline spec/models/sms/adapters/intelli_sms_adapter_spec.rb]: See above.

**Comment by hooverlunch** [comment]
Great work. Just a few changes. I can handle these myself when merging, so no more work needed on your end. Thanks!

**Comment by linesarefuzzy** [comment]
Cool, thanks!

**Pull request closed without merging**